### PR TITLE
mk_repo_file: Parallelize fetch

### LINKED ISSUE #169: Parallelise fetchers
I just tried updating LOS but GH is painfully slow at serving the repos sparsely (one kernel source was in the KiB/s...).

The scripts need to be parallelised, especially now that we've got 3 "sub-flavours" to boot.

Is there an easy way to parallelise a for loop in python?

### PR DESCRIPTION
**Pull request by zhaofengli**
Got impatient waiting for the 12.1 tags to sync. I didn't time the before/after runs but verified that it was indeed parallelized.

Fixes #169.

### CODE REVIEW

File: scripts/mk_repo_file.py [hunk 1]
```diff
@@ -232,6 +248,7 @@ def main() -> None:
     parser.add_argument('--include-prefix', action="append", default=[],
                         help="only include paths if they start with the specified prefix")
     parser.add_argument('--exclude-path', action="append", default=[], help="paths to exclude from fetching")
+    parser.add_argument('--jobs', '-j', default=16, type=int, help="number of concurrent jobs")
```
Comment (samueldr) [inline scripts/mk_repo_file.py:251]: Can this default to a detected number of cores?

Comment (Kranzes) [reply scripts/mk_repo_file.py:251]: This should definitely be auto-detected.

File: scripts/mk_repo_file.py [hunk 2]
```diff
@@ -191,8 +204,11 @@ def make_repo_file(url: str, ref: str,
 
             add_to_cache(p)
 
-            if callback is not None:
-                callback(data)
+            with cb_lock:
+                if callback is not None:
+                    callback(data)
+
+    pool.map(process_item, data.items())
```
Comment (hmenke) [inline scripts/mk_repo_file.py:211]: I think `pool.map` only returns a generator. It has to be evaluated explicitly using, e.g. `list`.

```suggestion
    list(pool.map(process_item, data.items()))
```

File: scripts/mk_repo_file.py (hunk 1 above)
Comment (Kranzes) [reply scripts/mk_repo_file.py:251]: Perhaps something needs to be done in line 84 too

File: scripts/mk_repo_file.py (hunk 2 above)
Comment (zhaofengli) [reply scripts/mk_repo_file.py:211]: No, the lazy version is `Pool.imap`.

**Comment by Kranzes** [comment]
Can someone give the same treatment to the LineageOS-specific scripts?, `update_device_dirs.py` and maybe `update_device_metadata.py`.

**Pull request merged**